src/ui/components/Library/Team/View/TestRuns/TestRunTestPanel/TestRunTestPanel.tsx

```diff
@@ -0,0 +1,138 @@
+import { useContext, useMemo, useState } from "react";
+
+import Icon from "replay-next/components/Icon";
+import { TestRunTestWithRecordings } from "shared/test-suites/TestRun";
+
+import { TestSuitePanelMessage } from "../../TestSuitePanelMessage";
+import { useTestRunDetailsSuspends } from "../hooks/useTestRunDetailsSuspends";
+import { TestRunPanelWrapper } from "../TestRunPanelWrapper";
+import { TestRunsContext } from "../TestRunsContextRoot";
+import { ExecutionList } from "./ExecutionList";
+import styles from "./TestRunTestPanel.module.css";
+
+export function TestRunTestPanel() {
+  const { testId, filterTestsByText } = useContext(TestRunsContext);
+  const { testRunId, testRuns } = useContext(TestRunsContext);
+
+  const { groupedTests, tests } = useTestRunDetailsSuspends(testRunId);
+  const selectedSpecTests =
+    // Select tests that not filtered in second panel
+    tests
+      ?.filter(
+        t => filterTestsByText === "" || t.sourcePath.toLowerCase().includes(filterTestsByText)
+      )
+      ?.filter(t => t.testId === testId) ?? [];
```

Comment:
I was a little confused by the naming here, given "tests" and "test id" seems to imply (to me) that there would only ever be one match. I wonder if there's a clearer name we could pick for this?

That being said, when I see chained `.filter` statements, I usually think "we could write this to be more efficient" using a single loop:
```suggestion
tests?.filter((test) => {
  if (test.testId !== testId) {
    return false;
  } else if (filterTestsByText !== "") {
    if (!test.sourcePath.toLowerCase().includes(filterTestsByText)) {
      return false;
    }
  }
  return true;
}) ?? [];
```